pipeline/resources/postgis_geopandas_io_manager.py

```diff
@@ -71,14 +71,21 @@ def handle_output(self, context: OutputContext, obj: pandas.DataFrame):
         if isinstance(obj, pandas.DataFrame):
             with connect_postgresql(config=self._config) as con:
                 self._create_schema_if_not_exists(schema, con)
-                # just recreate table with empty frame (obj[:0]) and load later via copy_from
-                obj[:0].to_sql(
-                    con=con,
-                    name=table,
-                    index=True,
-                    schema=schema,
-                    if_exists='replace',
-                )
+                table_exists = self._has_table(con, schema, table)
+                if table_exists:
+                    self._truncate_table(schema, table, con)
+                else:
+                    # create table with empty frame (obj[:0]) and load later via copy_from
+                    obj[:0].to_sql(
+                        con=con,
+                        name=table,
+                        index=True,
+                        schema=schema,
+                        if_exists='replace',
+                    )
+                    # table was just created, create primary key (to_postgis doesn't create these,
```

Comment:
We're calling `to_sql()` above, not `to_postgis()`. Can you expand the comment to explain what's going on?

Reply:
Added a comment that explains that `GeoDataFrame`'s methods are not named in a db agnostic as `pandas.DataFrame`'s `to_sql`.